### PR TITLE
Add e4s HIP build CI

### PR DESCRIPTION
**Pull request by streeve**

### CODE REVIEW

File: .github/workflows/CI-e4s.yml [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+jobs:
+  HIP:
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      matrix:
+        cxx: ['hipcc']
+        cmake_build_type: ['Debug', 'Release']
+        kokkos_ver: ['3.6.01']
```
Comment (dalg24) [inline .github/workflows/CI-e4s.yml:23]: Any good reason to use such an old version of Kokkos?

File: .github/workflows/CI-e4s.yml [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        with:
+          repository: google/googletest
+          ref: release-1.11.0
+          path: gtest
+      - name: Build gtest
+        working-directory: gtest
+        run: |
+          cmake -B build \
+            -DCMAKE_INSTALL_PREFIX=$HOME/gtest \
+            -DCMAKE_BUILD_TYPE=${{ matrix.cmake_build_type }}
```
Comment (dalg24) [inline .github/workflows/CI-e4s.yml:44]: I would always go for Release for GTest.

File: .github/workflows/CI-e4s.yml [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+          cmake -B build \
+            -DCMAKE_INSTALL_PREFIX=$HOME/kokkos \
+            -DCMAKE_CXX_COMPILER=${{ matrix.cxx }} \
+            -DCMAKE_BUILD_TYPE=${{ matrix.cmake_build_type }} \
+            -DKokkos_ENABLE_HIP=ON \
+            -DKokkos_ARCH_VEGA908=ON
+          cmake --build build --parallel 2
+          cmake --install build
+      - name: Checkout arborx
+        if: ${{ matrix.arborx == 'ArborX' }}
```
Comment (dalg24) [inline .github/workflows/CI-e4s.yml:65]: I thought it was not enabled when looking at L24

File: .github/workflows/CI-e4s.yml [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        with:
+          repository: arborx/ArborX
+          ref: v1.2
+          path: arborx
+      - name: Build arborx
+        if: ${{ matrix.arborx == 'ArborX' }}
+        working-directory: arborx
+        run: |
+          cmake -B build \
+            -DCMAKE_PREFIX_PATH=${HOME}/kokkos \
```
Comment (dalg24) [inline .github/workflows/CI-e4s.yml:76]: Works but prefer
```suggestion
            -DKokkos_ROOT=${HOME}/kokkos \
```

File: .github/workflows/CI-e4s.yml [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        if: ${{ matrix.heffte != 'OFF' }}
+        # actions/checkout doesn't work for external repos yet (actions/checkout#447)
+        run: |
+          git clone --depth 1 --branch v2.1.0 https://bitbucket.org/icl/heffte.git heffte
+      - name: Build heffte
+        if: ${{ matrix.heffte != 'OFF' }}
+        working-directory: heffte
+        run: |
+          cmake -B build \
+            -DCMAKE_CXX_STANDARD="11" \
```
Comment (dalg24) [inline .github/workflows/CI-e4s.yml:92]: Any good reason to enforce C++11 for heffte?

File: .github/workflows/CI-e4s.yml (hunk 1 above)
Comment (streeve) [reply .github/workflows/CI-e4s.yml]: Updated to 4.0

File: .github/workflows/CI-e4s.yml (hunk 2 above)
Comment (streeve) [reply .github/workflows/CI-e4s.yml]: Done

File: .github/workflows/CI-e4s.yml (hunk 3 above)
Comment (streeve) [reply .github/workflows/CI-e4s.yml]: Fixed

File: .github/workflows/CI-e4s.yml (hunk 4 above)
Comment (streeve) [reply .github/workflows/CI-e4s.yml]: Updated

File: .github/workflows/CI-e4s.yml (hunk 5 above)
Comment (streeve) [reply .github/workflows/CI-e4s.yml]: No - removed

**Comment by streeve** [comment]
@dalg24 this is ready for review, but the build is taking an hour - how can I restrict the architectures it builds for?

```
[  1%] Linking CXX executable Cabana_Version_test
cd /__w/Cabana/Cabana/build/core/unit_test && /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/cmake-3.26.3-3w3vsl2gab5bx2bunygxpalwywntzmoc/bin/cmake -E cmake_link_script CMakeFiles/Cabana_Version_test.dir/link.txt --verbose=1
/usr/bin/hipcc -Wall -Wextra -pedantic -Werror -g -DKOKKOS_DEPENDENCE -fno-gpu-rdc -Wl,-rpath -Wl,/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpich-4.1.1-4cbi7qhusseuuh6bcs6lokwgwh6s3itl/lib CMakeFiles/Cabana_Version_test.dir/tstVersion.cpp.o CMakeFiles/Cabana_Version_test.dir/__/__/cmake/test_harness/unit_test_main.cpp.o -o Cabana_Version_test  -Wl,-rpath,/spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpich-4.1.1-4cbi7qhusseuuh6bcs6lokwgwh6s3itl/lib /github/home/gtest/lib/libgtest.a /github/home/kokkos/lib/libkokkoscontainers.a /github/home/kokkos/lib/libkokkoscore.a -ldl /github/home/kokkos/lib/libkokkossimd.a /opt/rocm/lib/libamdhip64.so.5.4.50403 --hip-link --offload-arch=gfx900 --offload-arch=gfx906 --offload-arch=gfx908 --offload-arch=gfx90a --offload-arch=gfx1030 /opt/rocm-5.4.3/llvm/lib/clang/15.0.0/lib/linux/libclang_rt.builtins-x86_64.a /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpich-4.1.1-4cbi7qhusseuuh6bcs6lokwgwh6s3itl/lib/libmpicxx.so /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.1.0/mpich-4.1.1-4cbi7qhusseuuh6bcs6lokwgwh6s3itl/lib/libmpi.so 
```

**Comment by streeve** [comment]
> the build is taking an hour - how can I restrict the architectures it builds for?

Fixed the issue - @dalg24 @junghans this is ready for review

**Review by junghans — APPROVED** [review]

**Pull request merged**